python/feathub/processors/flink/flink_types_utils.py

```diff
@@ -92,6 +98,75 @@ def to_flink_type(feathub_type: DType) -> DataType:
     )
 
 
+def cast_field_type(
```

Comment:
According to Yuxia's comment on this jira ticket that links to FLINK-31830, the attempt to align the nullability behavior between SQL and Table API might be enrolled in discussions and design changes more complicated than we would expect from a bugfix. Thus we are still not sure whether we could remove/simplify this method in future.

Given the uncertainty above, would it be better to create an issue instead of adding TODOs or comments to the code? TODOs and comments would give me an impression that we had already got a determined migration plan for this method.

Reply:
Would it be useful to add a comment like this:

```
Use SQL instead of Table API to get around the issue in FLINK-32464
```

Also, would it be useful to rename this method as `cast_field_type_without_changing_nullability()`?